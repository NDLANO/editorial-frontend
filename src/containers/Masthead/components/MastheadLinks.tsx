/**
 * Copyright (c) 2024-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import { ComponentPropsWithoutRef } from "react";
import { SafeLink } from "@ndla/safelink";
import { styled } from "@ndla/styled-system/jsx";
import { JsxStyleProps } from "@ndla/styled-system/types";

const LinksContainer = styled("div", {
  base: {
    display: "flex",
    gap: "xsmall",
  },
});

export const MastheadLinks = (props: ComponentPropsWithoutRef<"div"> & JsxStyleProps) => {
  return (
    <LinksContainer {...props}>
      <SafeLink target="_blank" to="https://edndla.zendesk.com/hc/no">
        Zendesk
      </SafeLink>
      <SafeLink target="_blank" to="https://kvalitet.ndla.no/">
        Kvalitaisen
      </SafeLink>
    </LinksContainer>
  );
};
