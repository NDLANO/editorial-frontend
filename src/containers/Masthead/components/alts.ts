/**
 * Copyright (c) 2025-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

const alts = {
  nb: [
    "Norske Dovendyrs Laug og Allianse",
    "Nyttige Dagens Lunsj Anbefalinger",
    "Nasjonal Dugnad for Latter og Anarki",
    "Nysgjerrige Detektivers Løsnings Arkiv",
    "Nordmenns Daglige Lykke Aktivitet",
    "Nattens Dansende Lyskastere Allianse",
    "Nyttige Data for Lærings Aktiviteter",
    "Norges Dyreste Luksus Artikler",
    "Nydelige Dyr Lager Abstraksjoner",
    "Nasjonal Dugnad for Løft av Alpakkaer",
  ],
  nn: [
    "Norske Dovendyrs Laug og Allianse",
    "Nyttige Dagens Lunsj Anbefalinger",
    "Nasjonal Dugnad for Latter og Anarki",
    "Nysgjerrige Detektivers Løsnings Arkiv",
    "Nordmenns Daglige Lykke Aktivitet",
    "Nattens Dansende Lyskastere Allianse",
    "Nyttige Data for Lærings Aktiviteter",
    "Norges Dyreste Luksus Artikler",
    "Nydelige Dyr Lager Abstraksjoner",
    "Nasjonal Dugnad for Løft av Alpakkaer",
  ],
  en: [
    "Never Drink Lemonade Alone",
    "Nocturnal Dancing Llama Association",
    "Ninja Donuts Love Adventures",
    "New Dinosaur Language Academy",
    "Notorious Duck Lovers Anonymous",
    "National Day of Lazy Activities",
    "Nerds Discussing Lizard Anatomy",
    "Neverending Dreams of Lush Adventures",
    "Nomadic Dragons Learning Algebra",
    "Nostalgic Doodles and Laughs Assembly",
  ],
};
export default alts;
