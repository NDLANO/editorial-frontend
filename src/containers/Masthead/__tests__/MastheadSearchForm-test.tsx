/**
 * Copyright (c) 2017-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import nock from "nock";
import { ReactNode } from "react";
import { MemoryRouter } from "react-router-dom";
import { render } from "@testing-library/react";
import { taxonomyApi } from "../../../config";
import IntlWrapper from "../../../util/__tests__/IntlWrapper";
import { MastheadSearchForm } from "../components/MastheadSearchForm";

const noop = () => {};

const wrapper = (component: ReactNode) => (
  <MemoryRouter>
    <IntlWrapper>{component}</IntlWrapper>
  </MemoryRouter>
);

test.skip("MastheadSearchForm redirects on ndla url paste with id at the end", () => {
  const historyMock = {
    push: vi.fn(),
  };

  const { container } = render(
    wrapper(
      <MastheadSearchForm query="" onSearchQuerySubmit={noop} onChange={noop} setQuery={noop} setMenuOpen={noop} />,
    ),
  );
  expect(container).toMatchSnapshot();
  setTimeout(() => {
    expect(historyMock.push).toBeCalledTimes(1);
    expect(historyMock.push).toBeCalledWith("/subject-matter/learning-resource/3333/edit/nb");
  });
});

test.skip("MastheadSearchForm redirects on ndla url paste with taxonomy id at the end", () => {
  const historyMock = {
    push: vi.fn(),
  };

  nock("http://ndla-api")
    .get(`${taxonomyApi}/topics/urn:topic:1:179373/?language=nb`)
    .reply(200, { contentUri: "urn:content:4232" });

  const { container } = render(
    wrapper(
      <MastheadSearchForm
        query="https://ndla-frontend.test.api.ndla.no/article/urn:subject:100/urn:topic:1:179373"
        onChange={noop}
        onSearchQuerySubmit={noop}
        setMenuOpen={noop}
        setQuery={noop}
      />,
    ),
  );
  expect(container).toMatchSnapshot();
  setTimeout(() => {
    expect(historyMock.push).toBeCalledTimes(1);
    expect(historyMock.push).toBeCalledWith("/subject-matter/topic-article/4232/edit/nb");
  });
});

test.skip("MastheadSearchForm redirects on old ndla url paste with new id", () => {
  const historyMock = {
    push: vi.fn(),
  };

  nock("http://ndla-api").get("/draft-api/v1/drafts/external_id/4737").reply(200, { id: "123" });

  const { container } = render(
    wrapper(
      <MastheadSearchForm
        query="https://ndla.no/nb/node/4737?fag=36"
        onChange={noop}
        onSearchQuerySubmit={noop}
        setMenuOpen={noop}
        setQuery={noop}
      />,
    ),
  );
  expect(container).toMatchSnapshot();
  setTimeout(() => {
    expect(historyMock.push).toBeCalledTimes(1);
    expect(historyMock.push).nthCalledWith(0, "/subject-matter/learning-resource/123/edit/nb");
  });
});

test.skip("MastheadSearchForm invalid id at the end of the url", () => {
  const historyMock = {
    push: vi.fn(),
  };

  const { container } = render(
    wrapper(
      <MastheadSearchForm
        onChange={noop}
        query="https://ndla-frontend.test.api.ndla.no/article/urn:subject:100/urn:topic:1:179373/urn:resource:1:16838"
        onSearchQuerySubmit={noop}
        setMenuOpen={noop}
        setQuery={noop}
      />,
    ),
  );
  expect(container).toMatchSnapshot();
  setTimeout(() => {
    expect(historyMock.push).not.toBeCalledTimes(1);
  });
});

test.skip("MastheadSearchForm redirects on ndla node id pasted", () => {
  const historyMock = {
    push: vi.fn(),
  };
  nock("http://ndla-api").get("/draft-api/v1/drafts/external_id/4737").reply(200, { id: "123" });

  const { container } = render(
    wrapper(
      <MastheadSearchForm
        query="#4737"
        onChange={noop}
        onSearchQuerySubmit={noop}
        setMenuOpen={noop}
        setQuery={noop}
      />,
    ),
  );
  expect(container).toMatchSnapshot();
  setTimeout(() => {
    expect(historyMock.push).toBeCalledTimes(1);
    expect(historyMock.push).toBeCalledWith("/subject-matter/learning-resource/123/edit/nb");
  });
});
