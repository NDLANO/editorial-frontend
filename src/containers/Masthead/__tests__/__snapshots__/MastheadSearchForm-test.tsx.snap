// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MastheadSearchForm invalid id at the end of the url 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  border-radius: 4px;
  border: 1px solid transparent;
  font-size: 16;
  font-size: 0.8888888888888888rem;
  line-height: 1.2;
  padding: 6px;
  height: 48px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: all 100ms ease-in-out;
  transition: all 100ms ease-in-out;
  min-width: 180px;
}

.emotion-0:where([lang='zh'], .emotion-0[lang='zh-Hans'], [lang='zh-Hant']):not([data-pinyin]) {
  font-size: calc(16 * 1.11);
  font-size: calc(0.8888888888888888rem * 1.11);
}

.emotion-0:focus-within {
  background: #fff;
  border: 1px solid #a5bcd3;
}

.emotion-0:hover {
  cursor: pointer;
  border: 1px solid #a5bcd3;
}

.emotion-0 input {
  padding: 6px;
  background: transparent;
  border: 0;
  outline: none;
  color: #20588f;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.emotion-0 input:not(:focus-within)::-webkit-input-placeholder {
  color: #20588f;
  font-weight: 600;
  opacity: 1;
}

.emotion-0 input:not(:focus-within)::-moz-placeholder {
  color: #20588f;
  font-weight: 600;
  opacity: 1;
}

.emotion-0 input:not(:focus-within):-ms-input-placeholder {
  color: #20588f;
  font-weight: 600;
  opacity: 1;
}

.emotion-0 input:not(:focus-within)::placeholder {
  color: #20588f;
  font-weight: 600;
  opacity: 1;
}

.emotion-0>button {
  color: #777777;
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  gap: 6px;
  color: #fff;
  background: #20588f;
  border: 2px solid #20588f;
  border-radius: 4px;
  outline-width: 0;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: 'Source Sans Pro',Helvetica,Arial,STKaiti,'华文楷体',KaiTi,SimKai,'楷体',KaiU,DFKai-SB,'標楷體',SongTi,'宋体',sans-serif;
  font-weight: 600;
  -webkit-transition: all .2s cubic-bezier(0.17, 0.04, 0.03, 0.94);
  transition: all .2s cubic-bezier(0.17, 0.04, 0.03, 0.94);
  box-shadow: none;
  text-align: center;
  padding: 4px 12px;
  font-size: 16px;
  font-size: 0.8888888888888888rem;
  line-height: 1.5;
  min-height: 40px;
  padding: 0;
  border-radius: 0;
  color: inherit;
  font-size: inherit;
  background-color: transparent;
  box-shadow: none;
  border: none;
  font-weight: 400;
}

.emotion-1:hover,
.emotion-1:focus-visible {
  color: #20588f;
  background-color: #ceddea;
  border-color: #ceddea;
}

.emotion-1[disabled] {
  color: #777777;
  background-color: #e6e6e6;
  border-color: transparent;
  cursor: not-allowed;
}

.emotion-1:where([lang='zh'], .emotion-1[lang='zh-Hans'], [lang='zh-Hant']):not([data-pinyin]) {
  font-size: calc(16px * 1.11);
  font-size: calc(0.8888888888888888rem * 1.11);
}

.emotion-1:hover,
.emotion-1:active,
.emotion-1:disabled,
.emotion-1:focus-visible {
  box-shadow: none;
  color: #20588f;
  background-color: transparent;
  border: none;
}

.emotion-1:focus-visible {
  outline-width: medium;
}

.emotion-2 {
  margin: 0 12px;
  width: 24px;
  height: 24px;
}

.emotion-3 {
  display: inline-block;
  fill: currentColor;
  vertical-align: middle;
  line-height: 1em;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 16px;
  height: 16px;
}

<div>
  <form
    class="emotion-0"
  >
    <button
      class="emotion-1"
      type="submit"
    >
      <svg
        aria-hidden="true"
        class="emotion-2 emotion-3 emotion-4"
        data-icon=""
        data-license="Apache License 2.0"
        data-source="Material Design"
        fill="currentColor"
        preserveAspectRatio="xMidYMid meet"
        style="color: rgb(32, 88, 143);"
        viewBox="0 0 48 48"
      >
        <path
          d="M31 28h-1.6l-.5-.6a13 13 0 1 0-1.4 1.4l.5.6V31l10 10 3-3zm-12 0a9 9 0 1 1 0-18 9 9 0 0 1 0 18"
        />
      </svg>
    </button>
    <input
      placeholder="Søk"
      type="text"
      value="https://ndla-frontend.test.api.ndla.no/article/urn:subject:100/urn:topic:1:179373/urn:resource:1:16838"
    />
  </form>
</div>
`;

exports[`MastheadSearchForm redirects on ndla node id pasted 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  border-radius: 4px;
  border: 1px solid transparent;
  font-size: 16;
  font-size: 0.8888888888888888rem;
  line-height: 1.2;
  padding: 6px;
  height: 48px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: all 100ms ease-in-out;
  transition: all 100ms ease-in-out;
  min-width: 180px;
}

.emotion-0:where([lang='zh'], .emotion-0[lang='zh-Hans'], [lang='zh-Hant']):not([data-pinyin]) {
  font-size: calc(16 * 1.11);
  font-size: calc(0.8888888888888888rem * 1.11);
}

.emotion-0:focus-within {
  background: #fff;
  border: 1px solid #a5bcd3;
}

.emotion-0:hover {
  cursor: pointer;
  border: 1px solid #a5bcd3;
}

.emotion-0 input {
  padding: 6px;
  background: transparent;
  border: 0;
  outline: none;
  color: #20588f;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.emotion-0 input:not(:focus-within)::-webkit-input-placeholder {
  color: #20588f;
  font-weight: 600;
  opacity: 1;
}

.emotion-0 input:not(:focus-within)::-moz-placeholder {
  color: #20588f;
  font-weight: 600;
  opacity: 1;
}

.emotion-0 input:not(:focus-within):-ms-input-placeholder {
  color: #20588f;
  font-weight: 600;
  opacity: 1;
}

.emotion-0 input:not(:focus-within)::placeholder {
  color: #20588f;
  font-weight: 600;
  opacity: 1;
}

.emotion-0>button {
  color: #777777;
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  gap: 6px;
  color: #fff;
  background: #20588f;
  border: 2px solid #20588f;
  border-radius: 4px;
  outline-width: 0;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: 'Source Sans Pro',Helvetica,Arial,STKaiti,'华文楷体',KaiTi,SimKai,'楷体',KaiU,DFKai-SB,'標楷體',SongTi,'宋体',sans-serif;
  font-weight: 600;
  -webkit-transition: all .2s cubic-bezier(0.17, 0.04, 0.03, 0.94);
  transition: all .2s cubic-bezier(0.17, 0.04, 0.03, 0.94);
  box-shadow: none;
  text-align: center;
  padding: 4px 12px;
  font-size: 16px;
  font-size: 0.8888888888888888rem;
  line-height: 1.5;
  min-height: 40px;
  padding: 0;
  border-radius: 0;
  color: inherit;
  font-size: inherit;
  background-color: transparent;
  box-shadow: none;
  border: none;
  font-weight: 400;
}

.emotion-1:hover,
.emotion-1:focus-visible {
  color: #20588f;
  background-color: #ceddea;
  border-color: #ceddea;
}

.emotion-1[disabled] {
  color: #777777;
  background-color: #e6e6e6;
  border-color: transparent;
  cursor: not-allowed;
}

.emotion-1:where([lang='zh'], .emotion-1[lang='zh-Hans'], [lang='zh-Hant']):not([data-pinyin]) {
  font-size: calc(16px * 1.11);
  font-size: calc(0.8888888888888888rem * 1.11);
}

.emotion-1:hover,
.emotion-1:active,
.emotion-1:disabled,
.emotion-1:focus-visible {
  box-shadow: none;
  color: #20588f;
  background-color: transparent;
  border: none;
}

.emotion-1:focus-visible {
  outline-width: medium;
}

.emotion-2 {
  margin: 0 12px;
  width: 24px;
  height: 24px;
}

.emotion-3 {
  display: inline-block;
  fill: currentColor;
  vertical-align: middle;
  line-height: 1em;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 16px;
  height: 16px;
}

<div>
  <form
    class="emotion-0"
  >
    <button
      class="emotion-1"
      type="submit"
    >
      <svg
        aria-hidden="true"
        class="emotion-2 emotion-3 emotion-4"
        data-icon=""
        data-license="Apache License 2.0"
        data-source="Material Design"
        fill="currentColor"
        preserveAspectRatio="xMidYMid meet"
        style="color: rgb(32, 88, 143);"
        viewBox="0 0 48 48"
      >
        <path
          d="M31 28h-1.6l-.5-.6a13 13 0 1 0-1.4 1.4l.5.6V31l10 10 3-3zm-12 0a9 9 0 1 1 0-18 9 9 0 0 1 0 18"
        />
      </svg>
    </button>
    <input
      placeholder="Søk"
      type="text"
      value="#4737"
    />
  </form>
</div>
`;

exports[`MastheadSearchForm redirects on ndla url paste with id at the end 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  border-radius: 4px;
  border: 1px solid transparent;
  font-size: 16;
  font-size: 0.8888888888888888rem;
  line-height: 1.2;
  padding: 6px;
  height: 48px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: all 100ms ease-in-out;
  transition: all 100ms ease-in-out;
  min-width: 180px;
}

.emotion-0:where([lang='zh'], .emotion-0[lang='zh-Hans'], [lang='zh-Hant']):not([data-pinyin]) {
  font-size: calc(16 * 1.11);
  font-size: calc(0.8888888888888888rem * 1.11);
}

.emotion-0:focus-within {
  background: #fff;
  border: 1px solid #a5bcd3;
}

.emotion-0:hover {
  cursor: pointer;
  border: 1px solid #a5bcd3;
}

.emotion-0 input {
  padding: 6px;
  background: transparent;
  border: 0;
  outline: none;
  color: #20588f;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.emotion-0 input:not(:focus-within)::-webkit-input-placeholder {
  color: #20588f;
  font-weight: 600;
  opacity: 1;
}

.emotion-0 input:not(:focus-within)::-moz-placeholder {
  color: #20588f;
  font-weight: 600;
  opacity: 1;
}

.emotion-0 input:not(:focus-within):-ms-input-placeholder {
  color: #20588f;
  font-weight: 600;
  opacity: 1;
}

.emotion-0 input:not(:focus-within)::placeholder {
  color: #20588f;
  font-weight: 600;
  opacity: 1;
}

.emotion-0>button {
  color: #777777;
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  gap: 6px;
  color: #fff;
  background: #20588f;
  border: 2px solid #20588f;
  border-radius: 4px;
  outline-width: 0;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: 'Source Sans Pro',Helvetica,Arial,STKaiti,'华文楷体',KaiTi,SimKai,'楷体',KaiU,DFKai-SB,'標楷體',SongTi,'宋体',sans-serif;
  font-weight: 600;
  -webkit-transition: all .2s cubic-bezier(0.17, 0.04, 0.03, 0.94);
  transition: all .2s cubic-bezier(0.17, 0.04, 0.03, 0.94);
  box-shadow: none;
  text-align: center;
  padding: 4px 12px;
  font-size: 16px;
  font-size: 0.8888888888888888rem;
  line-height: 1.5;
  min-height: 40px;
  padding: 0;
  border-radius: 0;
  color: inherit;
  font-size: inherit;
  background-color: transparent;
  box-shadow: none;
  border: none;
  font-weight: 400;
}

.emotion-1:hover,
.emotion-1:focus-visible {
  color: #20588f;
  background-color: #ceddea;
  border-color: #ceddea;
}

.emotion-1[disabled] {
  color: #777777;
  background-color: #e6e6e6;
  border-color: transparent;
  cursor: not-allowed;
}

.emotion-1:where([lang='zh'], .emotion-1[lang='zh-Hans'], [lang='zh-Hant']):not([data-pinyin]) {
  font-size: calc(16px * 1.11);
  font-size: calc(0.8888888888888888rem * 1.11);
}

.emotion-1:hover,
.emotion-1:active,
.emotion-1:disabled,
.emotion-1:focus-visible {
  box-shadow: none;
  color: #20588f;
  background-color: transparent;
  border: none;
}

.emotion-1:focus-visible {
  outline-width: medium;
}

.emotion-2 {
  margin: 0 12px;
  width: 24px;
  height: 24px;
}

.emotion-3 {
  display: inline-block;
  fill: currentColor;
  vertical-align: middle;
  line-height: 1em;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 16px;
  height: 16px;
}

<div>
  <form
    class="emotion-0"
  >
    <button
      class="emotion-1"
      type="submit"
    >
      <svg
        aria-hidden="true"
        class="emotion-2 emotion-3 emotion-4"
        data-icon=""
        data-license="Apache License 2.0"
        data-source="Material Design"
        fill="currentColor"
        preserveAspectRatio="xMidYMid meet"
        style="color: rgb(32, 88, 143);"
        viewBox="0 0 48 48"
      >
        <path
          d="M31 28h-1.6l-.5-.6a13 13 0 1 0-1.4 1.4l.5.6V31l10 10 3-3zm-12 0a9 9 0 1 1 0-18 9 9 0 0 1 0 18"
        />
      </svg>
    </button>
    <input
      placeholder="Søk"
      type="text"
      value=""
    />
  </form>
</div>
`;

exports[`MastheadSearchForm redirects on ndla url paste with taxonomy id at the end 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  border-radius: 4px;
  border: 1px solid transparent;
  font-size: 16;
  font-size: 0.8888888888888888rem;
  line-height: 1.2;
  padding: 6px;
  height: 48px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: all 100ms ease-in-out;
  transition: all 100ms ease-in-out;
  min-width: 180px;
}

.emotion-0:where([lang='zh'], .emotion-0[lang='zh-Hans'], [lang='zh-Hant']):not([data-pinyin]) {
  font-size: calc(16 * 1.11);
  font-size: calc(0.8888888888888888rem * 1.11);
}

.emotion-0:focus-within {
  background: #fff;
  border: 1px solid #a5bcd3;
}

.emotion-0:hover {
  cursor: pointer;
  border: 1px solid #a5bcd3;
}

.emotion-0 input {
  padding: 6px;
  background: transparent;
  border: 0;
  outline: none;
  color: #20588f;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.emotion-0 input:not(:focus-within)::-webkit-input-placeholder {
  color: #20588f;
  font-weight: 600;
  opacity: 1;
}

.emotion-0 input:not(:focus-within)::-moz-placeholder {
  color: #20588f;
  font-weight: 600;
  opacity: 1;
}

.emotion-0 input:not(:focus-within):-ms-input-placeholder {
  color: #20588f;
  font-weight: 600;
  opacity: 1;
}

.emotion-0 input:not(:focus-within)::placeholder {
  color: #20588f;
  font-weight: 600;
  opacity: 1;
}

.emotion-0>button {
  color: #777777;
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  gap: 6px;
  color: #fff;
  background: #20588f;
  border: 2px solid #20588f;
  border-radius: 4px;
  outline-width: 0;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: 'Source Sans Pro',Helvetica,Arial,STKaiti,'华文楷体',KaiTi,SimKai,'楷体',KaiU,DFKai-SB,'標楷體',SongTi,'宋体',sans-serif;
  font-weight: 600;
  -webkit-transition: all .2s cubic-bezier(0.17, 0.04, 0.03, 0.94);
  transition: all .2s cubic-bezier(0.17, 0.04, 0.03, 0.94);
  box-shadow: none;
  text-align: center;
  padding: 4px 12px;
  font-size: 16px;
  font-size: 0.8888888888888888rem;
  line-height: 1.5;
  min-height: 40px;
  padding: 0;
  border-radius: 0;
  color: inherit;
  font-size: inherit;
  background-color: transparent;
  box-shadow: none;
  border: none;
  font-weight: 400;
}

.emotion-1:hover,
.emotion-1:focus-visible {
  color: #20588f;
  background-color: #ceddea;
  border-color: #ceddea;
}

.emotion-1[disabled] {
  color: #777777;
  background-color: #e6e6e6;
  border-color: transparent;
  cursor: not-allowed;
}

.emotion-1:where([lang='zh'], .emotion-1[lang='zh-Hans'], [lang='zh-Hant']):not([data-pinyin]) {
  font-size: calc(16px * 1.11);
  font-size: calc(0.8888888888888888rem * 1.11);
}

.emotion-1:hover,
.emotion-1:active,
.emotion-1:disabled,
.emotion-1:focus-visible {
  box-shadow: none;
  color: #20588f;
  background-color: transparent;
  border: none;
}

.emotion-1:focus-visible {
  outline-width: medium;
}

.emotion-2 {
  margin: 0 12px;
  width: 24px;
  height: 24px;
}

.emotion-3 {
  display: inline-block;
  fill: currentColor;
  vertical-align: middle;
  line-height: 1em;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 16px;
  height: 16px;
}

<div>
  <form
    class="emotion-0"
  >
    <button
      class="emotion-1"
      type="submit"
    >
      <svg
        aria-hidden="true"
        class="emotion-2 emotion-3 emotion-4"
        data-icon=""
        data-license="Apache License 2.0"
        data-source="Material Design"
        fill="currentColor"
        preserveAspectRatio="xMidYMid meet"
        style="color: rgb(32, 88, 143);"
        viewBox="0 0 48 48"
      >
        <path
          d="M31 28h-1.6l-.5-.6a13 13 0 1 0-1.4 1.4l.5.6V31l10 10 3-3zm-12 0a9 9 0 1 1 0-18 9 9 0 0 1 0 18"
        />
      </svg>
    </button>
    <input
      placeholder="Søk"
      type="text"
      value="https://ndla-frontend.test.api.ndla.no/article/urn:subject:100/urn:topic:1:179373"
    />
  </form>
</div>
`;

exports[`MastheadSearchForm redirects on old ndla url paste with new id 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  border-radius: 4px;
  border: 1px solid transparent;
  font-size: 16;
  font-size: 0.8888888888888888rem;
  line-height: 1.2;
  padding: 6px;
  height: 48px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transition: all 100ms ease-in-out;
  transition: all 100ms ease-in-out;
  min-width: 180px;
}

.emotion-0:where([lang='zh'], .emotion-0[lang='zh-Hans'], [lang='zh-Hant']):not([data-pinyin]) {
  font-size: calc(16 * 1.11);
  font-size: calc(0.8888888888888888rem * 1.11);
}

.emotion-0:focus-within {
  background: #fff;
  border: 1px solid #a5bcd3;
}

.emotion-0:hover {
  cursor: pointer;
  border: 1px solid #a5bcd3;
}

.emotion-0 input {
  padding: 6px;
  background: transparent;
  border: 0;
  outline: none;
  color: #20588f;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.emotion-0 input:not(:focus-within)::-webkit-input-placeholder {
  color: #20588f;
  font-weight: 600;
  opacity: 1;
}

.emotion-0 input:not(:focus-within)::-moz-placeholder {
  color: #20588f;
  font-weight: 600;
  opacity: 1;
}

.emotion-0 input:not(:focus-within):-ms-input-placeholder {
  color: #20588f;
  font-weight: 600;
  opacity: 1;
}

.emotion-0 input:not(:focus-within)::placeholder {
  color: #20588f;
  font-weight: 600;
  opacity: 1;
}

.emotion-0>button {
  color: #777777;
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  gap: 6px;
  color: #fff;
  background: #20588f;
  border: 2px solid #20588f;
  border-radius: 4px;
  outline-width: 0;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: 'Source Sans Pro',Helvetica,Arial,STKaiti,'华文楷体',KaiTi,SimKai,'楷体',KaiU,DFKai-SB,'標楷體',SongTi,'宋体',sans-serif;
  font-weight: 600;
  -webkit-transition: all .2s cubic-bezier(0.17, 0.04, 0.03, 0.94);
  transition: all .2s cubic-bezier(0.17, 0.04, 0.03, 0.94);
  box-shadow: none;
  text-align: center;
  padding: 4px 12px;
  font-size: 16px;
  font-size: 0.8888888888888888rem;
  line-height: 1.5;
  min-height: 40px;
  padding: 0;
  border-radius: 0;
  color: inherit;
  font-size: inherit;
  background-color: transparent;
  box-shadow: none;
  border: none;
  font-weight: 400;
}

.emotion-1:hover,
.emotion-1:focus-visible {
  color: #20588f;
  background-color: #ceddea;
  border-color: #ceddea;
}

.emotion-1[disabled] {
  color: #777777;
  background-color: #e6e6e6;
  border-color: transparent;
  cursor: not-allowed;
}

.emotion-1:where([lang='zh'], .emotion-1[lang='zh-Hans'], [lang='zh-Hant']):not([data-pinyin]) {
  font-size: calc(16px * 1.11);
  font-size: calc(0.8888888888888888rem * 1.11);
}

.emotion-1:hover,
.emotion-1:active,
.emotion-1:disabled,
.emotion-1:focus-visible {
  box-shadow: none;
  color: #20588f;
  background-color: transparent;
  border: none;
}

.emotion-1:focus-visible {
  outline-width: medium;
}

.emotion-2 {
  margin: 0 12px;
  width: 24px;
  height: 24px;
}

.emotion-3 {
  display: inline-block;
  fill: currentColor;
  vertical-align: middle;
  line-height: 1em;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 16px;
  height: 16px;
}

<div>
  <form
    class="emotion-0"
  >
    <button
      class="emotion-1"
      type="submit"
    >
      <svg
        aria-hidden="true"
        class="emotion-2 emotion-3 emotion-4"
        data-icon=""
        data-license="Apache License 2.0"
        data-source="Material Design"
        fill="currentColor"
        preserveAspectRatio="xMidYMid meet"
        style="color: rgb(32, 88, 143);"
        viewBox="0 0 48 48"
      >
        <path
          d="M31 28h-1.6l-.5-.6a13 13 0 1 0-1.4 1.4l.5.6V31l10 10 3-3zm-12 0a9 9 0 1 1 0-18 9 9 0 0 1 0 18"
        />
      </svg>
    </button>
    <input
      placeholder="Søk"
      type="text"
      value="https://ndla.no/nb/node/4737?fag=36"
    />
  </form>
</div>
`;
